### DISCUSSION TITLE
enhancement for stereo file

### DISCUSSION THREAD

**Discussion by jayantbansal21** [speechbrain/speechbrain#1915] [Q&A]
alright, So i am passing a wav audio file, which has stereo sound in it. I am using Speechbrain to reduce the noise I am using this "enhance_model = SpectralMaskEnhancement.from_hparams(
               source="speechbrain/metricgan-plus-voicebank",savedir="pretrained_models/mtl-mimic-voicebank", )" Okay. Now, this is generating a noise-reduced mono file.
i want a stereo noise-reduced file. Of course, i can just pass both channels separately and then combine them, but it's more time and resource-consuming. Therefore, is there a direct solution to it?

**Comment by pplantinga** [comment]
A simple way to change the code to process multi-channel audio is to change the loading function as below.

**NOTE:** The enhancement systems were designed for single-channel audio so they may not work as well when used this way.

```python
import torch
import torchaudio
from speechbrain.pretrained import SpectralMaskEnhancement

enhance_model = SpectralMaskEnhancement.from_hparams(
    source="speechbrain/metricgan-plus-voicebank",
    savedir="pretrained_models/metricgan-plus-voicebank",
)

# Load and add fake batch dimension
#noisy = enhance_model.load_audio("file.wav").unsqueeze(0)

# Alternative loading for multi-channel processing
noisy, sample_rate = torchaudio.load("file.wav")
assert sample_rate == 16000

# Add relative length tensor
enhanced = enhance_model.enhance_batch(noisy, lengths=torch.tensor([1.]))

# Saving enhanced signal on disk
torchaudio.save('enhanced.wav', enhanced.cpu(), 16000)
```
